lib/matplotlib/legend_handler.py

```diff
@@ -723,3 +724,77 @@ def create_artists(self, legend, orig_handle,
         self.update_prop(p, orig_handle, legend)
         p.set_transform(trans)
         return [p]
+
+
+class HandlerFancyArrowPatch(HandlerPatch):
+    """
+    Handler for FancyArrowPatch instances.
+    """
+    def _create_patch(self, legend, orig_handle, xdescent, ydescent, width,
+                      height, fontsize):
+        arrow = FancyArrowPatch([-xdescent,
+                                 -ydescent + height / 2],
+                                [-xdescent + width,
+                                 -ydescent + height / 2],
+                                mutation_scale=width / 3)
+        arrow.set_arrowstyle(orig_handle.get_arrowstyle())
+        return arrow
+
+
+class HandlerAnnotation(HandlerBase):
+    """
+    Handler for Annotation instances.
+    Defers to HandlerFancyArrowPatch to draw the annotation arrow (if any).
+    Parameters
+    ----------
+    pad : float, optional
+        If None, fall back to `legend.borderpad` asstr the default.
+        In units of fraction of font size.
+        Default is None.
+    width_ratios : tuple, optional
+        The relative width of the respective text/arrow legend annotation pair.
+        Must be of length 2.
+        Default is [1,4].
+    """
+
+    def __init__(
+        self,
+        pad=None,
+        width_ratios=[1, 4],
+        **kwargs
+    ):
+
+        self._pad = pad
+        self._width_ratios = width_ratios
+
+        HandlerBase.__init__(self, **kwargs)
+
+    def create_artists(
```

Comment:
Could the call aruguments be on multiple lines, but with a maximum line length of 79 characters? (if that makes sense; like
```python
def func(arg1, arg2, arg3, arg4,
         arg5)
```